nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py

```diff
@@ -141,3 +142,51 @@ def __exit__(self, *args):
 
     def __del__(self):
         self.vc.release()
+
+
+class MovieDataChunkIterator(GenericDataChunkIterator):
+    """DataChunkIterator specifically for use on RecordingExtractor objects."""
+
+    def __init__(
+        self,
+        movie_file: PathType,
+        buffer_gb: float = None,
+        buffer_shape: tuple = None,
+        chunk_mb: float = None,
+        chunk_shape: tuple = None,
+        stub: bool = False,
+    ):
+        self.video_capture_ob = VideoCaptureContext(movie_file)
+        if stub:
+            self.video_capture_ob.frame_count = 10
+        self._default_chunk_shape = False
+        if chunk_shape is None:
+            chunk_shape = (1, *self.video_capture_ob.get_frame_shape())
+            self._default_chunk_shape = True
+        super().__init__(
+            buffer_gb=buffer_gb,
+            buffer_shape=buffer_shape,
+            chunk_mb=chunk_mb,
+            chunk_shape=chunk_shape,
+            display_progress=True,
+        )
+        self._current_chunk = 1
+
+    def _get_data(self, selection: Tuple[slice]) -> Iterable:
+        if self._default_chunk_shape:
+            self._current_chunk += 1
+            frame = next(self.video_capture_ob)
+            return frame[np.newaxis, :]
+        frames_return = []
+        step = selection[0].step if selection[0].step is not None else 1
+        for frame_no in range(selection[0].start, selection[0].stop, step):
+            frame = self.video_capture_ob.get_movie_frame(frame_no)
+            frames_return.append(frame[selection[1:]])
+            self._current_chunk += 1
+        return np.concatenate(frames_return, axis=0)
```

Comment:
Since this is a private function we can make some stronger (and safer) assumptions about how the data is accessed and returned than we could if this was an all-purpose retrieval (like `recording.get_traces` is).

As such here are a couple suggestions:

i) The selections that will always feed into this come from https://github.com/hdmf-dev/hdmf/blob/dev/src/hdmf/data_utils.py#L237 - as such, the stuff about step sizes is unnecessary and worst case could allow the iterator to skip frames (data loss in writing into the final `h5py.Dataset`)

ii) Similar to how we did it in the integration of the context class with the movie interface, I would pre-allocate the numpy array so it already has a block in memory designated. Otherwise the `np.concatenate` at the end (or any new array casting) will double the literal amount of RAM taken up by the buffer (it will clear it quickly after filling and returning, but still could cap system usage).

iii) What is the `_current_chunk` counter for? I don't see it being utilized anywhere

iv) Similar to (i) we can make stronger assumptions about order of selections (namely that they are consecutive), and repeated calls to `self.video_capture_ob.get_movie_frame` result in unnecessary overhead via the calls to the `frame_count` setter (https://github.com/catalystneuro/nwb-conversion-tools/blob/6ad1d4211eeb0c4955130c9beb46bac40b8747df/nwb_conversion_tools/datainterfaces/behavior/movie/movie_utils.py#L102). I'd recommend just doing what you did for the default chunk logic and repeat the calling of `next` on the capture iterable, filling those frames into the corresponding part of the pre-allocated numpy array.